x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportUpdateFilterAction.java

```diff
@@ -0,0 +1,181 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.engine.VersionConflictEngineException;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.MlMetaIndex;
+import org.elasticsearch.xpack.core.ml.action.PutFilterAction;
+import org.elasticsearch.xpack.core.ml.action.UpdateFilterAction;
+import org.elasticsearch.xpack.core.ml.job.config.MlFilter;
+import org.elasticsearch.xpack.core.ml.job.messages.Messages;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.ml.job.JobManager;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.core.ClientHelper.ML_ORIGIN;
+import static org.elasticsearch.xpack.core.ClientHelper.executeAsyncWithOrigin;
+
+public class TransportUpdateFilterAction extends HandledTransportAction<UpdateFilterAction.Request, PutFilterAction.Response> {
+
+    private final Client client;
+    private final JobManager jobManager;
+
+    @Inject
+    public TransportUpdateFilterAction(Settings settings, ThreadPool threadPool,
+                                       TransportService transportService, ActionFilters actionFilters,
+                                       Client client, JobManager jobManager) {
+        super(settings, UpdateFilterAction.NAME, threadPool, transportService, actionFilters,
+                (Supplier<UpdateFilterAction.Request>) UpdateFilterAction.Request::new);
+        this.client = client;
+        this.jobManager = jobManager;
+    }
+
+    @Override
+    protected void doExecute(UpdateFilterAction.Request request, ActionListener<PutFilterAction.Response> listener) {
+        ActionListener<FilterWithVersion> filterListener = ActionListener.wrap(filterWithVersion -> {
+            updateFilter(filterWithVersion, request, listener);
+        }, listener::onFailure);
+
+        getFilterWithVersion(request.getFilterId(), filterListener);
+    }
+
+    private void updateFilter(FilterWithVersion filterWithVersion, UpdateFilterAction.Request request,
+                              ActionListener<PutFilterAction.Response> listener) {
+        MlFilter filter = filterWithVersion.filter;
+
+        if (request.isNoop()) {
+            listener.onResponse(new PutFilterAction.Response(filter));
+            return;
+        }
+
+        String description = filter.getDescription();
```

Comment:
`String description = request.getDescription() == null ? filter.getDescription() : request.getDescription()` for readability